src/iosrv/tests/drpc_client_tests.c

```diff
@@ -235,6 +236,76 @@ test_drpc_verify_notify_bio_error(void **state)
 	assert_int_equal(close_call_count, 1);
 }
 
+static void
+verify_notify_pool_svc_update(uuid_t pool_uuid, d_rank_list_t *svc_reps)
+{
+	Drpc__Call		*call;
+	Mgmt__ClusterEventReq	*req;
+	d_rank_list_t		*reps;
+	uuid_t			 puuid;
+	int			 reps_match;
+
+	call = drpc__call__unpack(NULL, sendmsg_msg_iov_len,
+				  sendmsg_msg_content);
+	assert_non_null(call);
+	assert_int_equal(call->module, DRPC_MODULE_MGMT);
+	assert_int_equal(call->method, DRPC_METHOD_MGMT_CLUSTER_EVENT);
+
+	/* Verify payload contents */
+	req = mgmt__cluster_event_req__unpack(NULL, call->body.len,
+					      call->body.data);
+	assert_non_null(req);
+	assert_int_equal(uuid_parse(req->event->pool_svc_info->pool_uuid,
+				    puuid), 0);
+	assert_int_equal(uuid_compare(puuid, pool_uuid), 0);
+	assert_int_equal(req->event->pool_svc_info->n_svc_reps,
+			 svc_reps->rl_nr);
+	reps = uint32_array_to_rank_list(req->event->pool_svc_info->svc_reps,
+					 svc_reps->rl_nr);
+	reps_match = daos_rank_list_identical(svc_reps, reps);
+	d_rank_list_free(reps);
+	assert_true(reps_match);
+
+	/* Cleanup */
+	mgmt__cluster_event_req__free_unpacked(req, NULL);
+	drpc__call__free_unpacked(call, NULL);
+}
+
+static void
+test_drpc_verify_notify_pool_svc_update(void **state)
+{
+	uuid_t		 pool_uuid;
+	uint32_t	 svc_reps[4] = {0, 1, 2, 3};
+	d_rank_list_t	*svc_ranks;
+
+	mock_valid_drpc_resp_in_recvmsg(DRPC__STATUS__SUCCESS);
+
+	assert_int_equal(drpc_init(), 0);
+
+	/* drpc connection created */
+	assert_int_equal(connect_sockfd, socket_return);
+
+	/* socket was left open */
+	assert_int_equal(close_call_count, 0);
+
+	/* Message was sent */
+	assert_non_null(sendmsg_msg_ptr);
```

Comment:
You can probably remove all this verification after drpc_init and drpc_fini - Guessing we test drpc_init behavior elsewhere. In practice I'm not sure you even need to init and fini in the unit tests, since it's all based on the mocks anyway.